dataflow/flex-templates/streaming_beam/streaming_beam.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+#
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An Apache Beam streaming pipeline example.
+
+It reads JSON encoded messages from Pub/Sub, transforms the message data and
+writes the results to BigQuery.
+"""
+
+from __future__ import absolute_import
+
+import argparse
+import json
+import logging
+import time
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+from apache_beam.options.pipeline_options import StandardOptions
+import apache_beam.transforms.window as window
+
+
+# Defines the BigQuery schema for the output table.
+SCHEMA = ('url:STRING,'
+          'num_reviews:INTEGER,'
+          'score:FLOAT64,'
+          'first_date:TIMESTAMP,'
+          'last_date:TIMESTAMP')
+
+
+class ParseJsonMessageDoFn(beam.DoFn):
+  """Parse the json encoded text message into a dict."""
+
+  def __init__(self):
+    beam.DoFn.__init__(self)
+
+  def process(self, element):
```

Comment:
We should try to avoid the generic `element` variable name and replace it for something more descriptive of the data it contains.

```py
def parse_json_message(message):
  row = json.loads(message)
  return {
    'url': row['url'],
    'score': 1.0 if row['review'] == 'positive' else 0.0,
    'processing_time': int(time.time()),
  }
```

Reply:
Done.